Exclude assembly files when WOLFSSL_ARMASM_INLINE is defined

rlm2002 opened the pull request:
# Description

If WOLFSSL_ARMASM_INLINE is defined with `cmake`, compile with inline `*_c.c`. Otherwise, use `.S` files in build.

Add `BUILD_ARM_THUMB` build option (enabled with WOLFSSL_ARMASM_THUMB2=yes). Compiles `thumb2-*` files from `wolfcrypt/src/port/arm` when enabled.

Reformatted file with Google Gemini. Tabs are 4 spaces, keep lines 80 characters or less.

zd#20314

# Testing

```
mkdir -p out
pushd out
cmake .. -DWOLFSSL_ARMASM_INLINE=yes
cmake --build .

grep -rin "DWOLFSSL_ARMASM_INLINE" .
```

rlm2002 committed:
add option for WOLFSSL_ARMASM_INLINE to CMake

rlm2002 committed:
additional check for ARM ASM Inline option

append thumb2 files, append inline c files with BUILD_ARMASM_INLINE

add all asm files. move curve25519 files under BUILD_CURVE25519

include remaining files

SparkiDev commented on cmake/functions.cmake:
```diff
@@ -749,9 +759,11 @@ function(generate_lib_src_list LIB_SOURCES)
-                                  wolfcrypt/src/port/arm/armv8-curve25519.S)
+                             if (NOT BUILD_ARMASM_INLINE)
+                                  list(APPEND LIB_SOURCES
+                                      wolfcrypt/src/port/arm/armv8-32-curve25519.S
```
There are armv8-32-curve25519_c.c and armv8-curve25519_c.c.
Add them to make.

SparkiDev commented on cmake/functions.cmake:
```diff
@@ -392,10 +393,14 @@ function(generate_lib_src_list LIB_SOURCES)
-                             wolfcrypt/src/port/arm/armv8-sha512-asm.S
-                             wolfcrypt/src/port/arm/armv8-32-sha512-asm.S)
+                        if(BUILD_ARMASM_INLINE)
+                             list(APPEND LIB_SOURCES wolfcrypt/src/port/arm/armv8-sha512.c)
```
Add armv8-sha512-asm_c.c and armv8-32-sha512-asm_c.c

SparkiDev commented on cmake/functions.cmake:
```diff
@@ -770,7 +782,10 @@ function(generate_lib_src_list LIB_SOURCES)
-                                  list(APPEND LIB_SOURCES wolfcrypt/src/port/arm/armv8-curve25519.S)
+                                  if(NOT BUILD_ARMASM_INLINE)
+                                      list(APPEND LIB_SOURCES 
+                                      wolfcrypt/src/port/arm/armv8-curve25519.S)
```
Add armv8-curve25519._c.c when inline.

rlm2002 committed:
formatting remove whitespace

format whitespace so tabs are 4 spaces

format character count to be 80 characters or less per line

remove bracket

dgarske commented on cmake/functions.cmake:
```diff
@@ -623,284 +752,351 @@ function(generate_lib_src_list LIB_SOURCES)
+                    list(APPEND LIB_SOURCES
+                        wolfcrypt/src/port/arm/thumb2-mlkem-asm_c.c)
+                elseif(BUILD_ARM_THUMB)
+                    list(APPEND LIB_SOURCES]
```
Looks like typo here? Notice `]`

rlm2002 replied:
Thanks. Change has been made.

dgarske commented:
Jenkins retest this please: "FIPSv2-regression'" failed (probably agent offline). History already lost (seems to only keep a few).
